### PR TITLE
feat: DE-3: Added schema for PR

### PR DESCRIPTION
**Pull request by pracheeSadwelkar**

### CODE REVIEW

**Review by sharmashalu — APPROVED** [review]

File: src/repositories/pullRequest/IDocument.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import IVersionalbleDocument from "../versionable/IDocument";
+export default interface IPullRequestDocument extends IVersionalbleDocument {
+    _id: string;
+    title: string;
+    memberId: string;
+    requestedReviewer: string;
```
Comment (Nikhil-Rawat) [inline src/repositories/pullRequest/IDocument.ts:6]: The requested reviewers will be an array of strings. 
**Note**: _It was not mentioned in the model. Soon will update it as well_

File: src/repositories/pullRequest/IDocument.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    createdAt: Date;
```
Comment (Nikhil-Rawat) [inline src/repositories/pullRequest/IDocument.ts:7]: Please replace createdAt with Pull_request_created_at.

File: src/repositories/pullRequest/IDocument.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    closedAt: Date;
```
Comment (Nikhil-Rawat) [inline src/repositories/pullRequest/IDocument.ts:8]: Change feild name -:
1. closedAt - closed_at
2. mergedAt = merged_at

File: src/repositories/pullRequest/Repository.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  ): Promise<IPullRequestDocument> {
+    return super.findById(query.id);
+  }
+
+  /**
+   * Create new PR
+   * @returns {pullRequest}
+   */
+  /* istanbul ignore next */
+  public async createPR(
```
Comment (Nikhil-Rawat) [inline src/repositories/pullRequest/Repository.ts:41]: assign an original id inside createPR method

File: src/repositories/pullRequest/Repository.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import * as mongoose from 'mongoose';
+import { pRModel } from './model';
+import VersioningRepository from '../versionable/Repository';
+import IPullRequestDocument from './IDocument'
+
+export default class PullRequestRepository extends VersioningRepository<IPullRequestDocument,
+  mongoose.Model<IPullRequestDocument>>{
+
```
Comment (Nikhil-Rawat) [inline src/repositories/pullRequest/Repository.ts:8]: create a generateId method.

File: src/repositories/pullRequest/Schema.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import VersionableSchema from "../versionable/Schema";
+
```
Comment (Nikhil-Rawat) [inline src/repositories/pullRequest/Schema.ts:3]: change the field name here as well

File: src/repositories/pullRequest/Repository.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    return super.update(doc);
+  }
+
+  /**
+   * Remove  pullRequest
+   * @returns {pullRequest}
+   */
+  /* istanbul ignore next */
+  public async deletePR(
+    doc: any,
```
Comment (Nikhil-Rawat) [inline src/repositories/pullRequest/Repository.ts:64]: use findOneandDelete method

File: src/repositories/pullRequest/IDocument.ts (hunk 1 above)
Comment (pracheeSadwelkar) [reply src/repositories/pullRequest/IDocument.ts:6]: Resolved

File: src/repositories/pullRequest/IDocument.ts (hunk 2 above)
Comment (pracheeSadwelkar) [reply src/repositories/pullRequest/IDocument.ts:7]: Resolved

File: src/repositories/pullRequest/IDocument.ts (hunk 3 above)
Comment (pracheeSadwelkar) [reply src/repositories/pullRequest/IDocument.ts:8]: resolved

File: src/repositories/pullRequest/Repository.ts (hunk 4 above)
Comment (pracheeSadwelkar) [reply src/repositories/pullRequest/Repository.ts:41]: done

File: src/repositories/pullRequest/Repository.ts (hunk 5 above)
Comment (pracheeSadwelkar) [reply src/repositories/pullRequest/Repository.ts:8]: done

File: src/repositories/pullRequest/Schema.ts (hunk 6 above)
Comment (pracheeSadwelkar) [reply src/repositories/pullRequest/Schema.ts:3]: done

**Review by Nikhil-Rawat — APPROVED** [review]

**Pull request merged**